libutils/json.c

```diff
@@ -296,34 +312,32 @@ static int JsonObjectCompare(
     int ret = JsonLength(a) - JsonLength(b);
     if (ret != 0)
     {
+        Log(LOG_LEVEL_DEBUG, "JsonObjectCompare() fails, length differs by %d", ret);
         return ret;
     }
 
-    JsonIterator iter_a = JsonIteratorInit(a);
-    JsonIterator iter_b = JsonIteratorInit(a);
+    JsonIterator iter = JsonIteratorInit(a);
 
-    for (size_t i = 0; i < JsonLength(a); i++)
+    while (JsonIteratorHasMore(&iter))
     {
-        const JsonElement *child_a = JsonIteratorNextValue(&iter_a);
-        const JsonElement *child_b = JsonIteratorNextValue(&iter_b);
-
-        const char *const key_a = JsonIteratorCurrentKey(&iter_a);
-        const char *const key_b = JsonIteratorCurrentKey(&iter_b);
+        const char *const key = JsonIteratorNextKey(&iter);
+        const JsonElement *const child_a = JsonObjectGet(a, key);
```

Comment:
This could use `JsonIteratorCurrentValue()` should be cheaper. But not a biggie.

Reply:
What if the keys were out-of-order between element a and b though? I think this should stay as-is.

Reply:
@craigcomstock You'd need to use `JsonObjectGet()` for `b`, but for `a`, you can use `JsonIteratorCurrentValue()` - it will give you the value corresponding to `key`, which you just got from `JsonIteratorNextKey()`.

Reply:
Right. Could just iterate over `a` , use `JsonIteratorCurrentValue()` and then check `JsonObjectGet()` for `b`.

Reply:
Exactly. It's just one less lookup.

Reply:
changed. thanks! :)